### DISCUSSION TITLE
[1.20.x] Build task fail with typescript error

### DISCUSSION THREAD

**Discussion by tmvkrpxl0** [cc-tweaked/CC-Tweaked#1806] [Q&A] [answered]
I've been following build instruction but it fails with typescript error:
```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at new NodeError (node:internal/errors:399:5)
    at validateString (node:internal/validators:163:11)
    at Object.join (node:path:1172:7)
    at <anonymous> (/tmp/CC-Tweaked/projects/web/src/htmlTransform/index.tsx:62:33) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```
File `CC-Tweaked/projects/web/src/htmlTransform/index.tsx` references `file.path` which does not exist on my pc:
https://github.com/cc-tweaked/CC-Tweaked/blob/550296edc5ab58373aebc0040dea95980be5d2d2/projects/web/src/htmlTransform/index.tsx#L59

To confirm this I have ran this command on node(v18.14.0)
```ts
await fs.readdir("/tmp/CC-Tweaked/projects/web/build/illuaminate", { withFileTypes: true, recursive: true }, function(err, file) { console.log(file.path + "") })
```
This prints "undefined". running it with `file` instead of `file.path` prints these entries instead:
```
Dirent { name: 'basic-terminal-45a33440.png', [Symbol(type)]: 1 
```

**Comment by tmvkrpxl0** [comment]
I've upgraded node to v20.12.2 and did the same test. The result was same.

**Reply by SquidDev** [reply, marked as answer]
We make use of the `recursive` argument to `fs.readdir`, which requires either NodeJS v18.17.0 or v20.1.0.

> I've upgraded node to v20.12.2 and did the same test. The result was same

Can you double check that the correct version of Node was on the path when doing that? This should definitely work. If so, can you run the following input in Node:

```
$ node
Welcome to Node.js v18.20.2.
Type ".help" for more information.
> (() => { const path = require("path"); require("fs").readdir(".github", {withFileTypes:true,recursive:true}, (x, e) => {for (const file of e) console.log(path.join(file.path, file.name)) }); })()
```

**Reply by tmvkrpxl0** [reply]
Yesterday I thought it was using node v20 when I did the 2nd test. but I opened up node and the version was v18.14.
now I actually changed version to v20 using nvm and this is the output:
```sh
$ node                                                20.12.2
Welcome to Node.js v20.12.2.
Type ".help" for more information.
> (() => { const path = require("path"); require("fs").readdir(".github", {withFileTypes:true,recursive:true}, (x, e) => {for (const file of e) console.log(path.join(file.path, file.name)) }); })()
.github/ISSUE_TEMPLATE
.github/matchers
.github/pull_request_template.md
.github/workflows
.github/ISSUE_TEMPLATE/bug_report.yaml
.github/ISSUE_TEMPLATE/config.yml
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/something_else.md
.github/matchers/checkstyle.json
.github/matchers/illuaminate.json
.github/matchers/junit.json
.github/workflows/main-ci.yml
.github/workflows/make-doc.sh
.github/workflows/make-doc.yml
undefined
```

**Comment by tmvkrpxl0** [comment]
Builds succeeds now :D
